ballerina/http_commons.bal

```diff
@@ -56,16 +58,47 @@ isolated function buildRequest(RequestMessage message) returns Request|ClientErr
     } else if (message is mime:Entity[]) {
         request.setBodyParts(message);
     } else {
-        var result = trap val:toJson(message);
-        if (result is error) {
-            return error InitializingOutboundRequestError("json conversion error: " + result.message(), result);
-        } else {
-            request.setJsonPayload(result);
+        match mediaType {
+            mime:APPLICATION_FORM_URLENCODED => {
+                string payload = check processUrlEncodedContent(message);
+                request.setTextPayload(payload, mime:APPLICATION_FORM_URLENCODED);
+            }
+            _ => {
+                json payload = check processJsonContent(message);
+                request.setJsonPayload(payload);
+            }
         }
     }
     return request;
 }
 
+isolated function processUrlEncodedContent(anydata message) returns string|ClientError {
+    if message is map<string> {
+        string[] messageParams = message.entries().toArray().'map(constructFormEntry);
+        string payload = strings:'join("&", ...messageParams);
+        string|error encodedContent = url:encode(payload, "UTF-8");
```

Comment:
I don't think what we have done here is correct. This encodes everything instead of just the value. Therefore the end result looks like this. For the below map.

```ballerina
map<string> message = {foo: "bar", zoo: "foo"};
```

```
foo%3Dbar%26zoo%3Dfoo
```
Instead of,
```
foo=bar&zoo=foo
```

Reply:
Fixed with [92bbefa2fa625f533dbb34cc3516a40fe9002f36](https://github.com/ballerina-platform/module-ballerina-http/pull/797/commits/92bbefa2fa625f533dbb34cc3516a40fe9002f36)